pkg/tee/rsa.go

```
@@ -0,0 +1,90 @@
+package tee
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+)
+
+// This is a wrapper package to ease out reading from certs that are generated with openssl.
+// The keys are generated with the following commands:
+// Private key:
+// openssl genrsa -out private.pem 2048
+// Public key:
+// openssl rsa -in private.pem -outform PEM -pubout -out public.pem
+
+// GenerateSignature generates a signature for the payload using the private key.
+func GenerateSignature(payload, privateKeyBytes []byte) ([]byte, error) {
+	// Hash the payload (this returns a [32]byte which we'll convert to a []byte later)
+	hash := sha256.Sum256(payload)
+
+	// Decode the key into a "block"
+	privateBlock, _ := pem.Decode(privateKeyBytes)
```

we ignore any errors?